compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/interop/JMethodResolver.java

```diff
@@ -272,8 +272,10 @@ private void validateExceptionTypes(JMethodRequest jMethodRequest, JMethod jMeth
         if ((throwsCheckedException && !jMethodRequest.returnsBErrorType) ||
                 (jMethodRequest.returnsBErrorType && !throwsCheckedException && !returnsErrorValue)) {
             throw new JInteropException(DiagnosticErrorCode.METHOD_SIGNATURE_DOES_NOT_MATCH,
-                    "No such Java method '" + jMethodRequest.methodName + "' which throws checked exception " +
-                            "found in class '" + jMethodRequest.declaringClass + "'");
+                    "Incompatible ballerina return type for the Java method '" + jMethodRequest.methodName +
+                            "' which throws checked exception found in class '" + jMethodRequest.declaringClass +
+                            "': expected '" + jMethodRequest.bReturnType + "|error', found '" +
```

Comment:
Can `jMethodRequest.bReturnType` have an error type? If so, this can lead to multiple `error` in the diagnostic message.

Reply:
Also we need to check jMethodRequest.bReturnType is nill, since we can directly shoe error

Reply:
> If `jMethodRequest.bReturnType` is error type or union of error, it will not result in this compilation error. Added some test cases regarding this.

Reply:
Addressed the suggestions with https://github.com/ballerina-platform/ballerina-lang/pull/33876/commits/825ec8d4ae49527d3e993b4aa9dd1969b5d6c23f